packages/web-frontend/src/utils/periodGranularities.js

```diff
@@ -259,19 +261,44 @@ const getDefaultDatesForRangeGranularities = (periodGranularity, defaultTimePeri
   return { startDate: moment(DEFAULT_MIN_DATE), endDate: moment() };
 };
 
-export function getDefaultDates(viewConfig) {
+export function getDefaultDateRangeToFetch(viewConfig) {
   const { periodGranularity, defaultTimePeriod } = viewConfig;
 
   // we need a valid granularity to proceed
   if (!periodGranularity) {
     return {};
   }
 
+  // Fetch data for specified range and determine default dates based on that
+  const { useDatesOfData, ...restOfDefaultTimePeriod } = defaultTimePeriod || {};
+  if (useDatesOfData) {
+    return getDefaultDatesForRangeGranularities(periodGranularity, restOfDefaultTimePeriod);
+  }
+
   const isSingleDate = GRANULARITIES_WITH_ONE_DATE.includes(periodGranularity);
   if (isSingleDate) {
-    return getDefaultDatesForSingleDateGranularities(periodGranularity, defaultTimePeriod);
+    return getDefaultDatesForSingleDateGranularities(periodGranularity, restOfDefaultTimePeriod);
+  }
+  return getDefaultDatesForRangeGranularities(periodGranularity, restOfDefaultTimePeriod);
+}
+
+export function getDefaultDatePickerDates(viewConfig) {
+  if (isNullishOrEmptyObject(viewConfig)) {
+    return { startDate: null, endDate: null };
   }
-  return getDefaultDatesForRangeGranularities(periodGranularity, defaultTimePeriod);
+
+  const {
+    defaultTimePeriod,
+    startDate: requestedStartDate,
+    endDate: requestedEndDate,
+    dataStartDate,
+    dataEndDate,
+  } = viewConfig;
```

Comment:
I really don't like that post-data-build fields exist on an object called `config`, which I would think should refer to the static configuration options that are defined in the db and available to web-frontend earlier on. That's a problem that exists long before your PR though, so just a mention without any need to do anything here

Reply:
Yeah good to point this out. Honestly wading through web-frontend proved to be the trickiest part of these changes by far. II think this area of the code is the one that has been most punished by our tendency to not quite deprecate legacy code 😅

I think a good refactor is in order, but out of the scope for this